ast/src/lang/parse/format.rs

```diff
@@ -54,7 +55,28 @@ impl Lang {
             }
             Ok(())
         })?;
-        Ok((cls, associations))
+
+        if self.lang.filter_by_implements() {
+            if let Some(implements_query) = self.lang.implements_query() {
+                let implements_q = self.lang.q(&implements_query, &NodeType::Class);
+                let tree = self.lang.parse(code, &NodeType::Class)?;
+                let mut cursoe = QueryCursor::new();
+                let mut matches = cursoe.matches(&implements_q, tree.root_node(), code.as_bytes());
```

Comment:
i wonder if we can directly query the Class node here, instead of the entire file

`node` instead of `tree.root_node()`

it should be much more efficient